akka-projection-core/src/main/scala/akka/projection/internal/OffsetStore.scala

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.internal
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import akka.Done
+import akka.projection.ProjectionId
+
+trait OffsetStore[Offset, T[_]] {
+  def readOffset(projectionId: ProjectionId)(implicit ec: ExecutionContext): Future[Option[Offset]]
+  def saveOffset(projectionId: ProjectionId, offset: Offset)(implicit ec: ExecutionContext): T[_]
+  def saveOffsetAsync(projectionId: ProjectionId, offset: Offset)(implicit ec: ExecutionContext): Future[Done]
+}
```

Comment:
Since the Kafka projection doesn't have its own `OffsetStore` we need to provide a contract to use one of another implementation.

I moved the `Offset` type parameter from `saveOffset` to the type level.

I created `saveOffsetAsync` to return a `Future` of `saveOffset` run by its underlying implementation.  @renatocaval also suggested providing a mechanism to lift an `OffsetStore` into another implementation that always returns a `Future[Done]`.

Reply:
I've removed the `OffsetStore` abstraction.